### PR TITLE
feat(496): algolia search

### LINKED ISSUE #496: Migrate to Algolia for search
**Technical Requirements**

- Lunr index should be disabled
- No affect on page performance
- UI Should look exactly as before

### PR DESCRIPTION
**Pull request by neo773**
Matches 1:1 with old UI where ever appropriate

https://github.com/user-attachments/assets/02bb3a8d-c7f8-415d-8b6e-95f645e845a6

/claim #496
/closes #496

### CODE REVIEW

**Comment by neo773** [comment]
@amitksingh1490 
Could I get a review

File: src/components/shared/Search/Results.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                "h-14 text-tailCall-gray bg-[#f1f1f1] hover:bg-tailCall-yellow  text-decoration-[none] scroll-m-2 hover:text-tailCall-gray p-2 my-1 rounded-lg flex flex-row items-center gap-x-4",
+                selectedIndex === index && "bg-tailCall-yellow font-semibold",
+              )}
+              ref={(el) => (itemRefs.current[index] = el)}
+            >
+              <ItemIcon type={item.type} />
+              <div className="flex flex-col">
+                <div
+                  className="text-[14px]"
+                  dangerouslySetInnerHTML={{__html: item?._snippetResult?.hierarchy?.lvl2?.value}}
```
Comment (amitksingh1490) [inline src/components/shared/Search/Results.tsx:54]: dangerouslySetInnerHTML is there a way to avoid this?

**Comment by tusharmath** [comment]
closing in favour of https://github.com/tailcallhq/tailcallhq.github.io/pull/499

**Pull request closed without merging**